### PR TITLE
Create Picklist Plugin

### LINKED ISSUE #335: Create picklist plugin
We need a new plugin: `vendure-plugin-picklist`. 

It's similar to the invoice plugin, with the important distinction that PDF's don't need to be saved. This should simplify things quite a bit.

What we'd like:
* Under `settings > picklist` is a screen similar to the invoices settings screen, where an admin can create templates for a picklist PDF
* On the order detail page, a button should be added `Download Picklist` that downloads the picklist PDF on the fly
* Nice to have: In the order detail page, it would be nice to be able to select multiple orders, to download picklists for multiple orders. This can become quite heavy, so it would be nice to do that in the worker, but how do we then send the PDF's back to the same request?
If the last point takes up too much time, we add it later

### PR DESCRIPTION
**Pull request by dalyathan**
# Description

The changes in this PR implement and close #335 
# Breaking changes

Does this PR include any breaking changes we should be aware of? **NO**

# Screenshots

Screenshot 2024-02-15 at 8 43 44 PM
Screenshot 2024-02-15 at 8 43 50 PM
Screenshot 2024-02-16 at 1 42 12 PM

# Checklist

📌 Always:
- [X] I have set a clear title
- [X] My PR is small and contains a single feature
- [X] I have [checked my own PR](## "Fix typo's and remove unused or commented out code")

👍 Most of the time:
- [X] I have added or updated test cases for important functionality
- [X] I have updated the README if needed

📦 For publishable packages:
- [X] I have [increased the version number](## "After increasing the version to the next patch/minor/major, the package will be published automatically after merge") in `package.json`
- [X] I have added my changes to the `CHANGELOG.md` [See this example](https://github.com/Pinelab-studio/pinelab-vendure-plugins/blob/main/packages/vendure-plugin-invoices/CHANGELOG.md)

## Summary by CodeRabbit

- **New Features**
    - Introduced the Vendure Picklist Plugin for creating and managing PDF picklists for orders.
    - Added new plugins for modifying customer orders, multiserver database session caching, facet suggestions, anonymized orders, admin components, and shipping extensions.
    - Implemented functionality for previewing, downloading individual, and bulk downloading picklists as PDF or zip files.
    - Provided GraphQL support for upserting and retrieving picklist configurations.
    - Enabled customization of picklist templates through an HTML editor in the admin UI.

- **Documentation**
    - Added comprehensive documentation for the Vendure Picklist Plugin, including setup instructions and usage guidelines.

- **Style**
    - Updated styles for the preview button in the picklist component.

- **Tests**
    - Introduced end-to-end tests for the Picklist Plugin, covering configuration, preview, and download functionalities.

### CODE REVIEW

**Review by martijnvdbrug — CHANGES_REQUESTED** [review]
Nice work already, but some changes are required still:

1. There are a lot of references to `invocies`. In the README and in the code. CTRl+F gives me 92 results for the word `invoice`. Picklist should be completely separate from the invoices plugin, so there shouldn't be any references to invoice
2. `Preview` and `Download` don't need different mutations and functions: It's all the same: We generate a PDF based on an order and return it. Picklists aren't stored, thats why preview and download are the same thing in this plugin.
3. `Enabled` isn't needed. Picklist plugin will always be enabled. There is no automatic generation on events or anything, so no need to be able to disable the plugin
4. There is an unused type `InvoicesListInput` in the grapphql schema. Also, the types, queries and mutations are referencing `1invoices`. This is a different plugin, and causes conflicts when a consumer will have both invoices and picklist plugin installed.

File: packages/vendure-plugin-picklist/src/api/picklist.service.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+import { defaultTemplate } from './default-template';
+import { InvoiceConfigEntity } from './invoice-config.entity';
+import { createReadStream, ReadStream } from 'fs';
+import { createTempFile, zipFiles, ZippableFile } from './file.util';
+import { SortOrder } from '@vendure/common/lib/generated-shop-types';
+import { InvoiceData } from './types';
+
+@Injectable()
+export class PicklistService {
+  jobQueue: JobQueue<{ channelToken: string; orderCode: string }> | undefined;
```
Comment (martijnvdbrug) [inline packages/vendure-plugin-picklist/src/api/picklist.service.ts:23]: We dont use the jobqueu right?

Comment (dalyathan) [reply packages/vendure-plugin-picklist/src/api/picklist.service.ts]: Yes, I will be removing those two lines

**Review by martijnvdbrug — APPROVED** [review]
Nice work thanks!

**Pull request merged**